src/api/transaction/settings.js

```diff
@@ -5,32 +5,23 @@ const _ = require('lodash');
 const utils = require('./utils');
 const ripple = utils.common.core;
 const validate = utils.common.validate;
-const constants = utils.common.constants;
-const InvalidRequestError = utils.common.errors.InvalidRequestError;
+const AccountFlagIndices = utils.common.constants.AccountFlagIndices;
+const AccountFields = utils.common.constants.AccountFields;
+const ValidationError = utils.common.errors.ValidationError;
 
 // Emptry string passed to setting will clear it
 const CLEAR_SETTING = '';
 
-/**
- * Set integer flags on a transaction based on input and a flag map
- *
- * @param {Transaction} transaction
- * @param {Object} input - Object whose properties determine whether
- *                to update the transaction's SetFlag or ClearFlag property
- * @param {Object} flags - Object that maps property names to transaction
- *                integer flag values
- *
- * @returns undefined
- */
-function setTransactionIntFlags(transaction, values, flags) {
-  for (let flagName in flags) {
+function setTransactionFlags(transaction, values) {
+  const nameToIndexMap = AccountFlagIndices;
+  for (let flagName in nameToIndexMap) {
```

Comment:
With the ripple-rest ( for lack of a better term atm) api? Does it purport
to allow setting more than one setting at once?

With that code, you couldn't. You could if you set all the settings you
wanted to do by bit setting with Flags.

Only AccountTxnID doesn't support the use of Flags to set it. It requires
the use of SetFlag.